File: fixtures/kotlin/src/main/kotlin/calculator/Calculator.kt
package calculator

class Calculator {
    private var lastResult: Double = 0.0

    fun add(a: Double, b: Double): Double {
        lastResult = Operations.add(a, b)
        return lastResult
    }

    fun subtract(a: Double, b: Double): Double {
        lastResult = Operations.subtract(a, b)
        return lastResult
    }

    fun multiply(a: Double, b: Double): Double {
        lastResult = Operations.multiply(a, b)
        return lastResult
    }

    fun divide(a: Double, b: Double): Double {
        lastResult = Operations.divide(a, b)
        return lastResult
    }

    fun getLastResult(): Double = lastResult
}

File: fixtures/dotnet/src/Calculator/Calculator.cs
namespace Calculator;

public class Calculator
{
    private double _lastResult;

    public double Add(double a, double b)
    {
        _lastResult = Operations.Add(a, b);
        return _lastResult;
    }

    public double Subtract(double a, double b)
    {
        _lastResult = Operations.Subtract(a, b);
        return _lastResult;
    }

    public double Multiply(double a, double b)
    {
        _lastResult = Operations.Multiply(a, b);
        return _lastResult;
    }

    public double Divide(double a, double b)
    {
        _lastResult = Operations.Divide(a, b);
        return _lastResult;
    }

    public double GetLastResult()
    {
        return _lastResult;
    }
}
